x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/deprecation/DeprecationInfoAction.java

```diff
@@ -132,7 +141,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 .array("node_settings", nodeSettingsIssues.toArray())
                 .field("index_settings")
                 .map(indexSettingsIssues)
-                .array("ml_settings", mlSettingsIssues.toArray())
+                .mapContents(pluginSettingsIssues)
```

Comment:
I see why you did it this way - it allows us to make this change without breaking the API response format - but I think it would be worth coordinating with the ES UI team to see what would work best for them, as I believe currently any new keys here will just be ignored by the UI. This API is consumed most commonly through the UI, so it would definitely be nice if it could pick up any new plugin issue categories automatically.

Reply:
@elastic/es-ui  any thoughts?

The idea here is that each plugin would have a unique key entry (i.e. how `ml_settings` is now). For `ml_settings` it will contain ALL deprecations related to ML.

Reply:
It looks like the Upgrade Assistant ends up [concatenating all of these arrays](https://github.com/elastic/kibana/blob/master/x-pack/plugins/upgrade_assistant/server/lib/es_migration_apis.ts#L67) into a single array anyway. As Gordon mentioned, the UI is currently designed to know about [all the available categories up-front](https://github.com/elastic/kibana/blob/master/src/core/server/elasticsearch/legacy/api_types.ts#L388), so it won't pick up new ones dynamically.

@alisonelizabeth is currently leading the 8.0 Upgrade Assistant effort and is best-situated to offer guidance on changes to the API design. I'll leave it to her to give you insight into how this information will be surfaced in the UI and how the API can change to best support that, if at all.

Reply:
> guidance on changes to the API design. 

Technically, this doesn't change the API at all. I just want to be able to add new ML deprecation warnings. If we could, please, @gwbrown @cjcenizal if there were to be API changes, can that be another PR that is handled by whomever owns the plugin?

Reply:
@benwtrent I’m comfortable with handling any additional changes in a separate PR since it doesn’t seem like there is an immediate impact. However, I’m on the same page as Gordon and CJ in that the UI won’t pick up any new plugin issue categories, and it doesn’t seem ideal to have to remember to add support in the UI whenever a new plugin category is added.

> I just want to be able to add new ML deprecation warnings.

Can you provide an example? Based on https://github.com/elastic/elasticsearch/pull/65284#discussion_r533687869, it sounds like all deprecations will remain under `ml_settings`, but just want to confirm.

I’ve also opened an issue on the Kibana side to track any follow-up work regarding this: https://github.com/elastic/kibana/issues/84761.

Reply:
> Can you provide an example? Based on #65284 (comment), it sounds like all deprecations will remain under ml_settings, but just want to confirm.

Yes.